pint_xarray/accessors.py

```diff
@@ -346,6 +462,108 @@ def dequantify(self):
+
+        Convert the data
+
+        >>> ds.pint.to({"a": "mm", "b": ureg.g})
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) int64 <Quantity([0 1 2 3 4], 'second')>
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([   0.  250.  500.  750. 1000.], 'millimet...
+            b        (x) float64 <Quantity([-1000.  -750.  -500.  -250.     0.], 'gra...
+        >>> ds.pint.to(a=ureg.mm, b="g")
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) int64 <Quantity([0 1 2 3 4], 'second')>
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([   0.  250.  500.  750. 1000.], 'millimet...
+            b        (x) float64 <Quantity([-1000.  -750.  -500.  -250.     0.], 'gra...
+
+        Convert coordinates
+
+        >>> ds.pint.to({"u": ureg.ms})
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([0.   0.25 0.5  0.75 1.  ], 'meter')>
+            b        (x) float64 <Quantity([-1.   -0.75 -0.5  -0.25  0.  ], 'kilogram')>
+        >>> ds.pint.to(u="ms")
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([0.   0.25 0.5  0.75 1.  ], 'meter')>
+            b        (x) float64 <Quantity([-1.   -0.75 -0.5  -0.25  0.  ], 'kilogram')>
+
+        Convert both simultaneously
+
+        >>> ds.pint.to(a=ureg.mm, b=ureg.g, u="ms")
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([   0.  250.  500.  750. 1000.], 'millimet...
+            b        (x) float64 <Quantity([-1000.  -750.  -500.  -250.     0.], 'gra...
+        >>> ds.pint.to({"a": "mm", "b": "g", "u": ureg.ms})
+        <xarray.Dataset>
+        Dimensions:  (x: 5)
+        Coordinates:
+            u        (x) float64 <Quantity([   0. 1000. 2000. 3000. 4000.], 'millisec...
+        Dimensions without coordinates: x
+        Data variables:
+            a        (x) float64 <Quantity([   0.  250.  500.  750. 1000.], 'millimet...
+            b        (x) float64 <Quantity([-1000.  -750.  -500.  -250.     0.], 'gra...
+        """
+        units = either_dict_or_kwargs(units, unit_kwargs, "to")
+
+        return conversion.convert_units(self.ds, units)
+
     def to_base_units(self):
```

Comment:
Likewise here?

Reply:
same here: not sure how to use `conversion`, but we should definitely add a possibility to only convert certain variables.